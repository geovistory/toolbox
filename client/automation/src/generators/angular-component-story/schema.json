{
  "$schema": "http://json-schema.org/schema",
  "$id": "AngularComponent",
  "title": "",
  "type": "object",
  "properties": {
    "componentPath": {
      "type": "string",
      "description": "Path to components folder relative to workspace root (e.g. './apps/app-toolbox/src/app/components/misc/new-button')",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Path to components folder relative to workspace root (e.g. './apps/app-toolbox/src/app/components/misc/new-button')"
    }
  },
  "required": [
    "componentPath"
  ]
}