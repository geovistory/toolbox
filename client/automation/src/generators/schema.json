{
  "$schema": "http://json-schema.org/schema",
  "$id": "AngularComponent",
  "title": "",
  "type": "object",
  "properties": {
    "componentFolder": {
      "type": "string",
      "description": "Relative path of the new component folder (e.g. 'apps/app-toolbox/src/app/components/misc/new-component')",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What is the path of the new component folder (e.g. 'apps/app-toolbox/src/app/components/misc/new-component')?"
    }
  },
  "required": [
    "componentFolder"
  ]
}