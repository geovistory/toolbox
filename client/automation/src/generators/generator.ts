import { componentGenerator } from '@nx/angular/generators';
import {
  formatFiles,
  names,
  Tree
} from '@nx/devkit';
import { AngularComponentGeneratorSchema } from './schema';

export async function angularComponentGenerator(
  tree: Tree,
  options: AngularComponentGeneratorSchema
) {

  const componentPath = options.componentFolder.replace(/\/+$/, '');
  const componentFolder = componentPath.split('/').pop();


  await componentGenerator(tree, {
    name: names(componentFolder).className,
    directory: componentPath,
    changeDetection: 'OnPush',
    displayBlock: true,
    prefix: 'gv',
    style: 'scss',
    skipTests: true
  })

  await formatFiles(tree);
}

export default angularComponentGenerator;
