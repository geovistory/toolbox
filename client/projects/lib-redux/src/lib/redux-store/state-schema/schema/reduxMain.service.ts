import { Injectable } from '@angular/core';
import { AccountDataService, AnalysisService, ContentTreeService, DataModelService, GvFieldPageReq, GvPaginationObject, GvPositiveSchemaObject, GvSchemaModifier, InfData, InfResourceWithRelations, InfStatementWithRelations, ProAnalysis, ProClassFieldConfig, ProDfhClassProjRel, ProInfoProjRel, ProjectConfigurationService, ProjectDataService, ProTextProperty, SubfieldPageControllerService } from '@kleiolab/lib-sdk-lb4';
import { Observable } from 'rxjs';
import { GvSchemaActions } from './schema.actions';

@Injectable({
  providedIn: 'root'
})
/**
 * Public API for all api calls returning
 * GvSchemaObject, GvNegativeSchemaObject, GvSchemaModifier
 */
export class ReduxMainService {

  constructor(
    protected schemaActions: GvSchemaActions,
    protected projectDataApi: ProjectDataService,
    protected projectConfigApi: ProjectConfigurationService,
    protected analysisApi: AnalysisService,
    protected dataModelApi: DataModelService,
    protected accountDataApi: AccountDataService,
    protected contentTree: ContentTreeService,
    protected pag: SubfieldPageControllerService,
  ) { }

  /**
   *
   * Sends rest api request to update ProInfoProjRel between pkProjeck and each entity in PkEntities,
   * to set is_in_project = false
   *
   * Merges returned GvSchmeObject into redux store.
   *
   * @param pkEntities
   * @param pkProject
   */
  removeInfEntitiesFromProject(pkEntities: number[], pkProject: number) {
    const call$ = this.projectDataApi.createProjectDataControllerUpsertInfoProjectRelations(
      pkProject,
      pkEntities.map((pk) => ({ fk_entity: pk, fk_project: pkProject, is_in_project: false }))
    )
    // bulkUpdateEprAttributes(
    //   pkProject,
    //   pkEntities.map((pk) => ({ fk_entity: pk, is_in_project: false }))
    // ) as Observable<ProInfoProjRel[]>
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  loadDatChunksOfDigital(pkDigital: number, pkProject: number): Observable<GvPositiveSchemaObject> {
    const call$ = this.projectDataApi.findProjectDataControllerGetChunksOfDigital(
      pkProject,
      pkDigital
    )
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  loadInfResource(pkResource: number, pkProject: number): Observable<GvPositiveSchemaObject> {
    const call$ = this.projectDataApi.findProjectDataControllerGetResource(
      pkProject,
      pkResource
    )
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  loadAllTypesOfProject(pkProject: number): Observable<GvPositiveSchemaObject> {
    const call$ = this.projectDataApi.findProjectDataControllerGetTypesOfProject(
      pkProject
    )
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  upsertInfResourcesWithRelations(pkProject: number, infResourceWithRelations: InfResourceWithRelations[]) {
    const call$ = this.projectDataApi.createProjectDataControllerUpsertResources(
      pkProject,
      infResourceWithRelations
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  upsertInfData(pkProject: number, infData: InfData) {
    const call$ = this.projectDataApi.createProjectDataControllerUpsertData(
      pkProject,
      infData
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  upsertInfStatementsWithRelations(pkProject: number, infStatementWithRelations: InfStatementWithRelations[])
    : Observable<GvSchemaModifier> {
    const call$ = this.projectDataApi.createProjectDataControllerUpsertStatements(
      pkProject,
      infStatementWithRelations
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  loadProjectsOfAccount(): Observable<GvPositiveSchemaObject> {
    const call$ = this.accountDataApi.accountDataControllerGetProjetcsOfAccount()
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  loadDfhProfilesOfProject(pkProject: number): Observable<GvSchemaModifier> {
    const call$ = this.dataModelApi.findDataModelControllerDfhProfilesOfProject(
      pkProject,
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  loadDfhPropertiesOfProject(pkProject: number): Observable<GvSchemaModifier> {
    const call$ = this.dataModelApi.dfhPropertyControllerOfProject(
      pkProject,
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }
  loadDfhClassesOfProject(pkProject: number): Observable<GvSchemaModifier> {
    const call$ = this.dataModelApi.findDataModelControllerDfhClassesOfProject(
      pkProject,
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }
  loadDfhLabelsOfProject(pkProject: number): Observable<GvSchemaModifier> {
    const call$ = this.dataModelApi.findDataModelControllerDfhLabelsOfProject(
      pkProject,
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  loadProjectBasics(pkProject: number): Observable<GvSchemaModifier> {
    const call$ = this.projectConfigApi.findProjectConfigControllerGetBasics(
      pkProject,
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  loadProjectConfiguration(pkProject: number): Observable<GvSchemaModifier> {
    const call$ = this.projectConfigApi.findProjectConfigControllerGetAllConfigsOfProject(
      pkProject,
    )
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  loadFieldPage(reqs: GvFieldPageReq[]): Observable<GvPaginationObject> {
    const call$ = this.pag.subfieldPageControllerLoadSubfieldPages(reqs)
    return this.schemaActions.loadGvPaginationObject(call$)
  }

  loadContentTree(pkProject: number, pkTreeRootEntity: number): Observable<GvPositiveSchemaObject> {
    const call$ = this.contentTree.contentTreeControllerGetContentTree(pkProject, pkTreeRootEntity)
    return this.schemaActions.loadGvSchemaObject(call$)
  }


  addEntityToProject(pkProject: number, pkEntity: number): Observable<GvPositiveSchemaObject> {
    const call$ = this.projectDataApi.addOrRemoveEntityControllerAddEntityToProject(pkProject, pkEntity)
    return this.schemaActions.loadGvSchemaObject(call$)
  }
  removeEntityFromProject(pkProject: number, pkEntity: number): Observable<GvPositiveSchemaObject> {
    const call$ = this.projectDataApi.addOrRemoveEntityControllerRemoveEntityFromProject(pkProject, pkEntity)
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  upsertInfoProjectRelations(pkProject: number, infoProjectRelations: Partial<ProInfoProjRel>[]): Observable<GvSchemaModifier> {
    const call$ = this.projectDataApi.createProjectDataControllerUpsertInfoProjectRelations(pkProject, infoProjectRelations)
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  upsertClassFieldConfig(pkProject: number, proClassFieldConfigs: Partial<ProClassFieldConfig>[]): Observable<GvPositiveSchemaObject> {
    const call$ = this.projectConfigApi.createProjectConfigControllerUpsertClassFieldConfigs(pkProject, proClassFieldConfigs)
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  loadProjectClassRelations(pkProject: number): Observable<GvPositiveSchemaObject> {
    const call$ = this.dataModelApi.findDataModelControllerClassProjectRelations(pkProject)
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  upsertProjectClassRelations(pkProject: number, proDfhClassProjRel: ProDfhClassProjRel[]): Observable<GvPositiveSchemaObject> {
    const call$ = this.projectConfigApi.createProjectConfigControllerUpsertProjectClassRelations(pkProject, proDfhClassProjRel)
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  loadProjectProfileRelations(pkProject: number): Observable<GvPositiveSchemaObject> {
    const call$ = this.dataModelApi.findDataModelControllerProfileProjectRelations(pkProject)
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  upsertProjectTextProperties(pkProject: number, proTextProperties: ProTextProperty[]): Observable<GvPositiveSchemaObject> {
    const call$ = this.projectConfigApi.createProjectConfigControllerPostTextProperties(pkProject, proTextProperties)
    return this.schemaActions.loadGvSchemaObject(call$)
  }

  deleteProjectTextProperties(pkProject: number, proTextProperties: ProTextProperty[]): Observable<GvSchemaModifier> {
    const call$ = this.projectConfigApi.createProjectConfigControllerDeleteTextProperties(pkProject, proTextProperties)
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  deleteProjectAnalisis(pkProject: number, pkEntities: number[]): Observable<GvSchemaModifier> {
    const call$ = this.analysisApi.analysisControllerBulkDelete(pkProject, pkEntities)
    return this.schemaActions.loadGvSchemaModifier(call$)
  }

  upsertProjectAnalisis(pkProject: number, proAnalysis: ProAnalysis[]): Observable<GvPositiveSchemaObject> {
    const call$ = this.analysisApi.analysisControllerBulkUpsert(pkProject, proAnalysis)
    return this.schemaActions.loadGvSchemaObject(call$)
  }
}
