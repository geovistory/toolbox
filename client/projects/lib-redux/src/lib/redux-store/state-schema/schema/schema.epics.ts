import { HttpErrorResponse } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Actions, concatLatestFrom, createEffect } from '@ngrx/effects';
import { Store } from '@ngrx/store';
import { FluxStandardAction } from 'flux-standard-action';
import { ofType } from 'redux-observable';
import { of } from 'rxjs';
import { catchError, mergeMap, startWith } from 'rxjs/operators';
import { IAppState } from '../../root/models/model';
import { NotificationsAPIActions } from '../../state-gui/actions/notifications.actions';
import { LoadingBarActions } from '../../state-gui/loadingbar/loading-bar.actions';
import { infStatementActions } from '../inf/statement/inf-statement.actions';
import { GvPaginationObjectAction, GvSchemaActions, GvSchemaModifierAction, GvSchemaObjectAction } from './schema.actions';
import { SchemaService } from './schema.service';

@Injectable({
  providedIn: 'root'
})
export class SchemaEpics {


  /**
   * Epic for loading GvSchemaObjects
   * - it subscribes to the given observable (payload), which usually triggers a REST API call
   * - on success it stores the GvSchemaObject
   * - else it toasts an error message
   */
  loadSchemaObject$ = createEffect(() => this.actions$.pipe(
    ofType(GvSchemaActions.GV_SCHEMA_OBJECT_LOAD),
    mergeMap((action: GvSchemaObjectAction) => action.payload.pipe(
      mergeMap(data => of(
        this.schemaObjectService.getStoreSchemaObjectGvAction(data),
        LoadingBarActions.REMOVE_JOB()
      )),
      catchError((error: HttpErrorResponse) => of(...this.errorActions(error))),
      startWith(LoadingBarActions.ADD_JOB()),
    ))
  ))
  /**
   * Epic for loading GvSchemaModifiers
   * - it subscribes to the given observable (payload), which usually triggers a REST API call
   * - on success it stores the GvSchemaObject
   * - else it toasts an error message
   */
  loadSchemaModifier$ = createEffect(() => this.actions$.pipe(
    ofType(GvSchemaActions.GV_SCHEMA_MODIFIER_LOAD),
    mergeMap((action: GvSchemaModifierAction) => action.payload.pipe(
      mergeMap(data => of(
        this.schemaObjectService.getStoreSchemaObjectGvAction(data.positive),
        ...this.schemaObjectService.getDeleteSchemaObjectGvActions(data.negative, 0),
        LoadingBarActions.REMOVE_JOB()
      )),
      catchError((error: HttpErrorResponse) => of(...this.errorActions(error))),
      startWith(LoadingBarActions.ADD_JOB()),
    ))
  ))

  /**
  * Epic for loading GvPaginationObjects
  * - it subscribes to the given observable (payload), which usually triggers a REST API call
  * - on success it stores the GvPaginationObject
  * - else it toasts an error message
  */
  loadPaginationObject$ = createEffect(() => this.actions$.pipe(
    ofType(GvSchemaActions.GV_PAGINATION_OBJECT_LOAD),
    mergeMap((action: GvPaginationObjectAction) => action.payload.pipe(
      concatLatestFrom(() => this.store.select((s) => s.activeProject?.pk_project)),
      mergeMap(([data, pkProject]) => {
        const pageLoadedActions = data.subfieldPages.map(p => infStatementActions.loadPageSucceededAction(
          p.paginatedStatements, p.count, p.req.page, pkProject
        ))
        return of(
          ...pageLoadedActions,
          LoadingBarActions.REMOVE_JOB()
        )
      }),
      catchError((error: HttpErrorResponse) => of(...this.errorActions(error))),
      startWith(LoadingBarActions.ADD_JOB()),
    ))
  ))


  constructor(
    private schemaObjectService: SchemaService,
    private notificationActions: NotificationsAPIActions,
    private actions$: Actions<FluxStandardAction<any, any>>,
    private store: Store<IAppState>
  ) { }


  errorActions(error: HttpErrorResponse) {
    return [
      LoadingBarActions.REMOVE_JOB(),
      this.notificationActions.addToast({
        type: 'error',
        options: {
          title: error.name,
          msg: error.message
        }
      })
    ]
  }
}

