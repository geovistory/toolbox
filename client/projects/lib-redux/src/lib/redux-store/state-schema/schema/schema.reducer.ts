import { GvPositiveSchemaObject, GvSchemaModifier } from '@kleiolab/lib-sdk-lb4';
import { FluxStandardAction } from 'flux-standard-action';
import { IAppState } from '../../root/models/model';
import { datDefinitions } from '../dat/dat.config';
import { infDefinitions } from '../inf/inf.config';
import { proDefinitions } from '../pro/pro.config';
import { dfhDefinitions } from '../reducer-configs/dfh.config';
import { tabDefinitions } from '../reducer-configs/tab.config';
import { sysDefinitions } from '../sys/sys.config';
import { warDefinitions } from '../war/war.config';
import { addToEntityModelMap, mergeItemsInState, ReducerConfigCollection } from '../_helpers/crud-reducer-factory';
import { GvSchemaActions } from './schema.actions';

const definitions = {
  dat: datDefinitions,
  dfh: dfhDefinitions,
  inf: infDefinitions,
  pro: proDefinitions,
  sys: sysDefinitions,
  tab: tabDefinitions,
  war: warDefinitions,
}

export const schemaModifierReducer = (state: IAppState = {}, action: FluxStandardAction<GvSchemaModifier>) => {

  if (action.type === GvSchemaActions.GV_MODIFIER_LOAD_SUCCEEDED) {
    if (action.payload.positive) {

      state = {
        ...state,
        ...loopOverSchemaNames(state, action.payload.positive)
      }
    }
    if (action.payload.negative) {
      // todo
    }

  }
  return state;
}


function loopOverSchemaNames(state: IAppState, positive: GvPositiveSchemaObject) {

  Object.keys(positive).forEach(schemaName => {
    const schemaDef: ReducerConfigCollection = definitions[schemaName];
    if (schemaDef) {
      const schemaData = positive[schemaName];
      const schemaState = state[schemaName] || {}
      state = {
        ...state,
        [schemaName]: {
          ...schemaState,
          ...loopOverModels(schemaData, schemaDef, schemaState)
        }
      }

    }

  });
  return state;


}


function loopOverModels(schemaData: any, schemaDef: ReducerConfigCollection, schemaState: any) {
  Object.keys(schemaData).forEach(modelName => {
    const modelDef = schemaDef[modelName];
    const modelData = schemaData[modelName];
    const modelState = schemaState[modelName] || {};
    if (modelDef) {
      schemaState = {
        ...schemaState,
        [modelName]: mergeItemsInState(modelDef, modelState, modelData),
        pkEntityModelMap: {
          ...schemaState.pkEntityModelMap,
          ...addToEntityModelMap(modelData, modelName)
        }
      }

    }
  });
  return schemaState
}
