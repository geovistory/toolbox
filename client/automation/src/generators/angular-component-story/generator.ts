import { componentStoryGenerator } from '@nx/angular/generators';
import {
  Tree,
  formatFiles,
  names
} from '@nx/devkit';
import { AngularComponentStoryGeneratorSchema } from './schema';

export async function angularComponentStoryGenerator(
  tree: Tree,
  options: AngularComponentStoryGeneratorSchema
) {

  const componentPath = options.componentPath
    .replace(/\/+$/, '') // remove trailing '/'
    .replace(/^\.\//, ''); // remove starting './'
  const pathSegments = componentPath.split('/');
  const componentFolder = pathSegments[pathSegments.length - 1];
  const componentName = names(componentFolder).className;
  const projectPath = './' + pathSegments.slice(0, 2).join('/');

  await componentStoryGenerator(tree, {
    componentFileName: componentFolder + '.component',
    componentName: componentName + 'Component',
    componentPath: pathSegments.slice(2).join('/'),
    projectPath: projectPath,
    interactionTests: true,
    skipFormat: false
  })
  await formatFiles(tree);
}

export default angularComponentStoryGenerator;
