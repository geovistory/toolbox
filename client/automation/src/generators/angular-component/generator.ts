import { componentGenerator } from '@nx/angular/generators';
import {
  Tree,
  formatFiles,
  names
} from '@nx/devkit';
import angularComponentStoryGenerator from '../angular-component-story/generator';
import { AngularComponentGeneratorSchema } from './schema';

export async function angularComponentGenerator(
  tree: Tree,
  options: AngularComponentGeneratorSchema
) {

  const componentPath = options.componentPath
    .replace(/\/+$/, '') // remove trailing '/'
    .replace(/^\.\//, ''); // remove starting './'
  const pathSegments = componentPath.split('/');
  const componentFolder = pathSegments[pathSegments.length - 1];
  const componentName = names(componentFolder).className;

  await componentGenerator(tree, {
    name: componentName,
    directory: './' + pathSegments.slice(0, -1).join('/'),
    export: false,
    standalone: true,
    changeDetection: 'OnPush',
    displayBlock: true,
    prefix: 'gv',
    style: 'scss',
    skipTests: true,
  })

  await angularComponentStoryGenerator(tree, options)

  await formatFiles(tree);
}

export default angularComponentGenerator;
