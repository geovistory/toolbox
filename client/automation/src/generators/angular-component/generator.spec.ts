import { Tree, addProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import { angularComponentGenerator } from './generator';
import { AngularComponentGeneratorSchema } from './schema';

describe('angular-component generator', () => {
  let tree: Tree;
  const componentFolder = `libs/lib1/src/lib/example`;
  const options: AngularComponentGeneratorSchema = { componentPath: componentFolder };

  beforeEach(async () => {
    // ARRANGE
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'lib1', {
      projectType: 'library',
      sourceRoot: 'libs/lib1/src',
      root: 'libs/lib1',
    });
  });

  it('should run successfully', async () => {
    await angularComponentGenerator(tree, options);
    // ASSERT
    expect(
      tree.read('libs/lib1/src/lib/example/example.component.ts', 'utf-8')
    ).toMatchSnapshot('component');
    expect(
      tree.read('libs/lib1/src/lib/example/example.component.html', 'utf-8')
    ).toMatchSnapshot('template');
    expect(
      tree.read('libs/lib1/src/lib/example/example.component.scss', 'utf-8')
    ).toMatchSnapshot('stylesheet');
    expect(
      tree.read('libs/lib1/src/lib/example/example.component.stories.ts', 'utf-8')
    ).toMatchSnapshot('stories');
  });
});
