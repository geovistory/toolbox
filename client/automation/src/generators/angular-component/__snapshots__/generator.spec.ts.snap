// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`angular-component generator should run successfully: component 1`] = `
"import { ChangeDetectionStrategy, Component } from '@angular/core';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'gv-example',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './example.component.html',
  styleUrls: ['./example.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ExampleComponent {}
"
`;

exports[`angular-component generator should run successfully: stories 1`] = `
"import {
  applicationConfig,
  type Meta,
  type StoryObj,
} from '@storybook/angular';
import { ExampleComponent } from './example.component';
import { importProvidersFrom } from '@angular/core';
import { MatDialogModule } from '@angular/material/dialog';
import { provideAnimations } from '@angular/platform-browser/animations';
import { StateModule } from '@kleiolab/lib-redux';
import { INITIAL_STATE } from '@ngrx/store';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

import { MockStateFactory } from './../../../.storybook/MockStateFactory';
import { ActiveProjectService } from './../services/active-project.service';

const meta: Meta<ExampleComponent> = {
  component: ExampleComponent,
  title: 'ExampleComponent',
  decorators: [
    applicationConfig({
      providers: [
        provideAnimations(),
        importProvidersFrom(StateModule, MatDialogModule),
        ActiveProjectService,
      ],
    }),
  ],
};
export default meta;
type Story = StoryObj<ExampleComponent>;

// Setup the initial state of the story Basic
const stateBasic = new MockStateFactory();

export const Basic: Story = {
  args: {},
  decorators: [
    applicationConfig({
      providers: [{ provide: INITIAL_STATE, useValue: stateBasic.state }],
    }),
  ],
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/example works!/gi)).toBeTruthy();
  },
};
"
`;

exports[`angular-component generator should run successfully: stylesheet 1`] = `
":host {
  display: block;
}
"
`;

exports[`angular-component generator should run successfully: template 1`] = `
"<p>Example works!</p>
"
`;
